odc/stac/_mdtools.py

```diff
@@ -681,6 +682,9 @@ def _get_grid(grid_name: str, asset: pystac.asset.Asset) -> GeoBox:
         asset = _assets.get(asset_name)
         if asset is None:
             continue
+        # the assets that aren't bands should be accessories
+        if asset_name in _acc_names:
+            _acc_names.remove(asset_name)
```

Comment:
Why start full then remove?
Let's start empty and add as we go instead

Reply:
Well I can't think of a better way of determining which assets are accessories other than by discounting the ones we know to be bands. Instead of `_acc_names` I could define a list of `band_names` and then do a set difference, but I don't think that's a cleaner solution than the current one.

Reply:
- `remove` is `O(n)`, repeated in a `O(n)` loop, so this is `O(n^2)` operation after this change

I'd rather this was a set operation, this also makes it easier to read to me, as it would be a single line that fully defines what `_acc_names` is, rather than currently where you start with "all asset names" then need to search code below to understand that this then becomes "all asset names except for data assets". Also `if asset_name in _acc_names:` will always be true in this logic.

Reply:
That's fair; updated.